CSharpFunctionalExtensions/Result/Extensions/Combine.cs

```diff
@@ -6,68 +6,75 @@ namespace CSharpFunctionalExtensions
 {
     public static partial class ResultExtensions
     {
-        public static Result Combine(this IEnumerable<Result> results, string errorMessageSeparator = null)
-            => Result.Combine(results, errorMessageSeparator);
-
-        public static Result<IEnumerable<T>, E> Combine<T, E>(this IEnumerable<Result<T, E>> results)
+        public static IResult Combine(this IEnumerable<IResult> results, string errorMessageSeparator = null)
+        {
+            results = results.ToList();
+            var first = results.First();
```

Comment:
it will throw is collection was empty, right?

The same thing with the other `.First()` calls